runtime/lua/vim/treesitter/_range.lua

```diff
@@ -194,4 +194,36 @@ function M.add_bytes(source, range)
   return { start_row, start_col, start_byte, end_row, end_col, end_byte }
 end
 
+---Compute intersection of a given range with the array of ranges.
+---Ranges are end-exclusive.
+---@private
+---@param it Range4
+---@param ranges Range4[] Sorted, non-intersecting.
+---@returns Range4[]
+function M.clip_range4_arr4(it, ranges)
```

Comment:
A table of `Range`s is a `Region`, and I think the naming should reflect that. (Looking forward, we'll need a proper abstraction for regions (e.g., block selections) anyway, so we should start laying the groundwork for that.)